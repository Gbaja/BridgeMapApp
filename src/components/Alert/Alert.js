import React from "react";

import {ERROR_ALERT} from "../../helpers/user_type";

function Alert({ alert: {message, type} }) { 
  return type == ERROR_ALERT && <div><p>{message}</p></div>;
}

export default Alert;

// having a function defined like it is above makes it easy to debug as it shows error message that includes the function name whereas with arrow function it treats the function as an anonymous function.
