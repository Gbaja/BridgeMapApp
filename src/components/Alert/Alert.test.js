import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Alert from "./Alert";

describe("App", () => {
  const alert = {
    type: "error",
    message: "heyy"
  };
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Alert alert={alert} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Alert alert={alert} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("shallow rendering", () => {
    const wrapper = shallow(<Alert alert={alert} />);
    expect(wrapper.find("p").length).toBe(1);
  });
  test('has a valid snapshot', ()=>{
    const alert = {
      type: "unknown"
    };
    const component = renderer.create(<Alert alert={alert} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
