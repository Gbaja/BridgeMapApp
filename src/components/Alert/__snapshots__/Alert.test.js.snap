// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div>
  <p>
    heyy
  </p>
</div>
`;

exports[`App has a valid snapshot 2`] = `null`;
